Link with Winsock 2

vadz opened the pull request:
See [#15562](https://trac.wxwidgets.org/ticket/15562).

vadz committed:
Remove obsolete libraries from .gitignore

Remove some Watcom and, apparently, Symantec-specific files as both
compilers are not used any more since many, many years.

vadz committed:
Remove obsolete and unnecessary wsock32.dll dynamic loading code

This is unnecessary at least since removal of Windows CE support back in
8282c1be0f (Remove Windows CE support, 2015-08-27) and probably for much
longer.

vadz committed:
Link with ws2_32.lib rather than obsolete wsock32.lib

Winsock 2 provided by ws2_32.lib is available since 1994, it should be
fine to use it even for wx in 2021.

See #15562.

vadz commented:
Any opinions about setting `wxUSE_WINSOCK2` to 1 by default?

Note to self: we should predefine `_WINSOCK_DEPRECATED_BY` if we do this to avoid warnings about `WSAAsyncSelect()` being deprecated in libcurl wxWebRequest backend.

PBfordev commented:
I do not know if you are planning it, but perhaps, to make life of application developers easier, CMake's FindwxWidgets module could be updated, just like it was done for example here [here](https://gitlab.kitware.com/cmake/cmake/-/commit/ae4b3b707204e7eecbb217395aa40a3c8b88aaa5).

However, I am not sure how to do that when it is version dependent (just link both?)...

Let's hope that the developers of the 3rd party tools such as Code::Blocks' template wizard or CodeLite's wx-config are going to update their tools as well.

vadz commented:
Thanks for the reminder! Unfortunately update CMake file is going to be trickier this time around, as we must select the right library to use depending on the version as I don't think we can link with both `wsock32.lib` and `ws2_32.lib`. And this file looks pretty intimidating, i.e. I don't immediately see how to check if the version of wx we found is 3.1.6 or greater in it... As usual, and especially with CMake, any help would be welcome.

vadz committed:
Enable Winsock 2 and IPv6 build options by default

In addition to linking with Winsock 2, also use winsock2.h.

This allows to turn on IPv6 support on by default under MSW, so turn it
on under Unix too for consistency.

Predefine _WINSOCK_DEPRECATED_NO_WARNINGS to avoid warnings about
deprecated (but still available and very unlikely to be ever removed)
Winsock 1 functions that we still use.
